responder/responder.go

```diff
@@ -39,8 +40,17 @@ func (r *Responder) JSON(ctx context.Context, w http.ResponseWriter, status int,
 	w.WriteHeader(status)
 
 	if _, err = w.Write(b); err != nil {
+		responseDescription := "response"
+		bodyLength := len(b)
+		if bodyLength > 1000 {
+			responseDescription = fmt.Sprintf("response clipped to 1000 chars from: %d", bodyLength)
+			bodyLength = 1000
+		}
 		log.Error(ctx, "failed to write response", err, log.Data{
-			"response": string(b),
+			// Limit length of the response that is logged to a useful amount to stop giving
+			// logstash a bad day as the dp-population-types-api has been seen to log a line
+			// that is ~1.9 M Bytes long - which is way too much.
+			responseDescription: string(b[0:bodyLength]),
```

Comment:
We really should be using dynamically generated field names. Any instance specific data should be in values rather than keys. 

Can I suggest…
```
	if _, err = w.Write(b); err != nil {
		var logData log.Data
		if len(b) > 1000 {
			// Limit length of the response that is logged to a useful amount to stop giving
			// logstash a bad day as the dp-population-types-api has been seen to log a line
			// that is ~1.9 M Bytes long - which is way too much.
                         logData = log.Data{
                             "truncated_response": string(b[:1000]),
                             "original_response_length": len(b),
                         }
		} else {
                         logData = log.Data{
                             "response": string(b),
                             "response_length": len(b),
                         }
		}   
		log.Error(ctx, "failed to write response", err, logData)
		return
	}
```

Reply:
done